PSKoans/Public/Show-Karma.ps1

```diff
@@ -4,26 +4,33 @@ function Show-Karma {
     [OutputType([void])]
     [Alias('Invoke-PSKoans', 'Test-Koans', 'Get-Enlightenment', 'Meditate', 'Clear-Path', 'Measure-Karma')]
     param(
+        [Parameter(ParameterSetName = 'ListKoans')]
+        [Parameter(ParameterSetName = 'ModuleOnly')]
+        [Parameter(ParameterSetName = 'IncludeModule')]
         [Parameter(ParameterSetName = 'Default')]
         [Alias('Koan', 'File')]
-        [ArgumentCompleter(
-            {
-                param($Command, $Parameter, $WordToComplete, $CommandAst, $FakeBoundParams)
-
-                $Values = (Get-PSKoanFile).Topic
-                return @($Values) -like "$WordToComplete*"
-            }
-        )]
         [SupportsWildcards()]
         [string[]]
         $Topic,
 
+        [Parameter(Mandatory, ParameterSetName = 'ModuleOnly')]
+        [Parameter(ParameterSetName = 'ListKoans')]
+        [SupportsWildcards()]
+        [string[]]
+        $Module,
+
+        [Parameter(Mandatory, ParameterSetName = 'IncludeModule')]
+        [Parameter(ParameterSetName = 'ListKoans')]
+        [SupportsWildcards()]
+        [string[]]
+        $IncludeModule,
```

Comment:
We probably shouldn't allow this to be called at the same time as `-Module`, even in the `ListKoans` parameter set. Might have to split that into two separate sets or something?

Reply:
Same for Get-Karma as well, of course. 🙂